nRF91 support without Partition Manager (#27973)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Board support
=============

> * Updated the Thingy91 and Thingy91x Device Tree partitions to match the Partition manager Partitions.

Reviewer: ```suggestion
* Updated the the Thingy:91 and Thingy:91 X devicetree partitions to match the Partition Manager partitions.
```

Binary libraries
----------------

* :ref:`liblwm2m_carrier_readme` library:

  * Added support for building without Partition Manager.

>     This requires an ``lwm2m_carrier`` partition to be defined in devicetree.

Reviewer 2: Doesn't need to be done in this PR, but this requirement should be in the `lwm2m_carrier` documentation together with instructions for how it should look.

Author: Doc update in #28318

Modem libraries
---------------

> * Added support for building for the nRF91 board without Partition Manager.

Reviewer 2: I think this belongs under a ```:ref:`nrf_modem_lib_readme`:``` header.

We should also have a similar note for the carrier library under `Binary libraries`, ```:ref:`liblwm2m_carrier_readme` library:```

  This requires having cpuapp_cpucell_ipc_shm_ctrl, cpuapp_cpucell_ipc_shm_heap and cpucell_cpuapp_ipc_shm_heap partitions in Device Tree.

> * Use cpucell_cpuapp_ipc_shm_trace partition for tracing instead of splitting the cpucell_cpuapp_ipc_shm_heap partition.

Reviewer: ```suggestion
  This requires ``cpuapp_cpucell_ipc_shm_ctrl``, ``cpuapp_cpucell_ipc_shm_heap``, and ``cpucell_cpuapp_ipc_shm_heap`` partitions to be defined in the devicetree.
  Use the ``cpucell_cpuapp_ipc_shm_trace`` partition for tracing instead of splitting the ``cpucell_cpuapp_ipc_shm_heap`` partition.
```

Author: I did make a separate point of the last line as this change also applies to the (unreleased) nRF92. I updated the line to hopefully make that more clear.

Modem libraries
---------------

* :ref:`nrf_modem_lib_readme` library:

  * Added support for building for the nRF91 board without Partition Manager.
    This requires ``cpuapp_cpucell_ipc_shm_ctrl``, ``cpuapp_cpucell_ipc_shm_heap``, and ``cpucell_cpuapp_ipc_shm_heap`` partitions to be defined in the devicetree.

>   * For both the nRF91 and the nRF92: Use a separate ``cpucell_cpuapp_ipc_shm_trace`` partition for tracing instead of splitting the ``cpucell_cpuapp_ipc_shm_heap`` partition.

Reviewer 3: ```suggestion
  * For both the nRF91 and the nRF92: Use a separate ``cpucell_cpuapp_ipc_shm_trace`` partition for modem tracing instead of splitting the ``cpucell_cpuapp_ipc_shm_heap`` partition.
```
